test/e2e/util/util.go

```diff
@@ -239,6 +239,27 @@ func WaitForConditionsOnHostedControlPlane(t *testing.T, ctx context.Context, cl
 	t.Logf("Observed hostedcluster to have successfully rolled out image. Namespace: %s, name: %s, image: %s", hostedCluster.Namespace, hostedCluster.Name, image)
 }
 
+// WaitForNodePoolVersion blocks until the NodePool status indicates the given
+// version. If the context is closed before the version is observed, the given
+// test will get an error.
+func WaitForNodePoolVersion(t *testing.T, ctx context.Context, client crclient.Client, nodePool *hyperv1.NodePool, version string) {
+	g := NewWithT(t)
+
+	t.Logf("Waiting for nodepool %s/%s to report version %s (currently %s)", nodePool.Namespace, nodePool.Name, version, nodePool.Status.Version)
+	err := wait.PollImmediateUntil(10*time.Second, func() (done bool, err error) {
+		latest := nodePool.DeepCopy()
+		err = client.Get(ctx, crclient.ObjectKeyFromObject(nodePool), latest)
+		if err != nil {
+			t.Logf("Failed to get nodepool: %v", err)
+			return false, nil
+		}
+		return latest.Status.Version == version, nil
```

Comment:
Can we please also validate here that `NodePoolUpdatingVersionConditionType` is not true?